src/main/java/edu/ucdavis/dss/ipa/api/components/course/CourseViewController.java

```diff
@@ -209,6 +215,96 @@ public Course addTagToCourse(@PathVariable long courseId, @PathVariable long tag
 		return courseService.addTag(course, tag);
 	}
 
+	@RequestMapping(value = "/api/courseView/workgroups/{workgroupId}/years/{year}/massAddTags", method = RequestMethod.PUT, produces="application/json")
+	@ResponseBody
+	public void massAddTagsToCourses(@PathVariable long workgroupId, @PathVariable long year, @RequestBody MassAssignTagsDTO massAssignTags, HttpServletResponse httpResponse) {
+		Schedule schedule = this.scheduleService.findByWorkgroupIdAndYear(workgroupId, year);
+		Workgroup workgroup = schedule.getWorkgroup();
+		Authorizer.hasWorkgroupRole(workgroup.getId(), "academicPlanner");
+
+		courseService.massAddTagsToCourses(massAssignTags.getTagIdsToAdd(), massAssignTags.getTagIdsToRemove(), massAssignTags.getCourseIds());
```

Comment:
Do we need to check that the body has anything? Is the DTO created if the body is empty?